### ISSUE TITLE
Allowing "get", "set" and "delete" as method/property names

### ISSUE THREAD

**Issue by ylazy** [airsdk/Adobe-Runtime-Support#3282]
I'm implementing Set and Map to AS3. Actually I've done it, but I have to rename the get/set/delete functions because it's reserved names.
Is it possible for ASC to allow using these names as method/property name?

Thanks!

**Comment by ajwfrost** [comment]
Just to clarify what you mean, you would like to be able to have something like:
```
public class Map
{
    public function set(key:*, value:*) : void
    {
        // stuff
    }
}
```

I think the compiler might be challenged to differentiate that from:
```
public class Map
{
    public function set myval(key:*, value:*) : void
    {
        // stuff
    }
}
```
which is a property setter..

Is there any reason you couldn't use capitalised versions or something to differentiate it from the reserved words? I'd be a little hesitant about changing those kinds of things..

thanks

**Comment by ylazy** [comment]
Yes, that's exactly what I mean. I'm also want to use some other reserved keywords like `break`, `continue`, `default`...
The reason is, in some case I think they're the best names to use.
In above example, I just want my classes (Map/Set) to have same API compared to JavaScript, so anyone that familiar with javascript can use my lib right away.
[Set.delete()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/delete)
[Map.delete()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete)
[Map.get()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get)
[Map.set()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set)

Another example, I have a class `AsyncLoop` that can construct something like for loop. It can handle multiple items need to be loaded/processed asynchronously. And I'm also want some functions called `break`, `continue`.

I'm also building some library that use command line processes. The command line APIs have flags/switches named `include`, `default`... And I want to use them as property names.

In some programing languages, there're ways to use reserved names. For example in C# I can use [the `@` prefix](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/).

```as
public class Map
{
    public function set(key:*, value:*) : void
    {
        // if there's nothing after set, so set is the function name
    }
}
public class Map
{
    public function set myval(key:*, value:*) : void
    {
        // there's a something after set (myval), so set is a keyword
    }
}
```

**Comment by ylazy** [comment]
`try`, `catch`, `finally` should be available for method names:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/try
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Currently I'm using `attempt`, `otherwise`, `always`.
